### PR TITLE
Add remove task

### PR DESCRIPTION
**Pull request by gabrie-lhilarion**
I added ability to add, delete and update a task
To edit/update a task click on the task description text

### CODE REVIEW

File: src/data.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    localStorage.setItem('mytodoTasks', JSON.stringify([]));
+    return [];
+  }
+
+  static updateData(status, index) {
+    const { allTasks: storedData } = Data;
+    storedData[Number(index)].completed = status;
+
+    localStorage.setItem('mytodoTasks', JSON.stringify(storedData));
+  }
```
Comment (henatan99) [inline src/data.js:19]: - [ ] When I check a todo item after the first batch of todo items are created, it updates the completed status of the next todo item in the local storage. But after page refresh, it works perfectly. I think you have an indexing problem when the page loads initially. Kindly fix this issue. :pray: 

Screenshot from 2021-12-05 02-15-27

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @gabrie-lhilarion , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/ui.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        <input type="text" id="task-description" name="description" placeholder="Add to your list...">
+         
+        <span class="save-task">&crarr;</span>
+    </li>
+    `;
+  }
+
+  static todoList(tasks) {
+    return tasks
+      .map((task, index) => `
```
Comment (rloterh) [inline src/ui.js:22]: - [ ] New entries **do not match** the property index. The _property index_ must be set to the value of the _new array length_ (i.e. if a 5th task is added to the list, the index of that task should equal 5). Kindly ensure to improve upon this aspect

**Index mismatch:**
image

File: src/handlers.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    const { allTasks, create, resetData } = Data;
+    const currentIndex = document.querySelectorAll('.task-item').length;
+
+    const description = e.target.previousElementSibling.value;
+    if (description.length < 3) return;
+    const newTask = create(description, currentIndex, completed);
+
+    const afterAddNew = [...allTasks, newTask];
+
+    // afterAddNew.forEach((item, index) => { item.index = index; });
```
Comment (rloterh) [inline src/handlers.js:24]: - [ ] Kindly remove commented code, so as to make your code base well organized

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Gabriel, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- CRUD function is working as intended
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gabrie-lhilarion** [comment]
# CORRECTED

- Indexing error pointed out by the last reviewer orrected

**Review by dasileker — APPROVED** [review]
## **Hi @gabrie-lhilarion,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**